### PR TITLE
[Merged by Bors] - Added olm package for 23.4.0-rc1

### PR DESCRIPTION
**Pull request by razvan**
Created OLM package for the upcomming 23.4.0-rc1.

New: metadata/dependencies.yaml to specify operator dependencies.

This is what OS shows when dependencies are not installed:
image

### CODE REVIEW

File: deploy/olm/23.4.0-rc1/manifests/csv.yaml [hunk 1]
```diff
@@ -0,0 +1,226 @@
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
```
Comment (adwk67) [inline deploy/olm/23.4.0-rc1/manifests/csv.yaml:3]: I would rename this file to something with `clusterserviceversion.yaml` in the suffix, as this is checked for in the certification pipeline (not needed in the bundle, but might help us not to forget if we name it here already).

File: deploy/olm/23.4.0-rc1/manifests/csv.yaml [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+    # The following affect how the package is indexed at OperatorHub.io:
+    # https://operatorhub.io/?category=Database
+    # https://sdk.operatorframework.io/docs/advanced-topics/operator-capabilities/operator-capabilities/
+    categories: Storage
+    capabilities: Full Lifecycle
+    description: Apache Zookeeper for the Stackable platform
+
+
+  description: Stackable operator for Apache Zookeeper
```
Comment (adwk67) [inline deploy/olm/23.4.0-rc1/manifests/csv.yaml:19]:
```suggestion
  description: |-
    This is a Kubernetes operator to manage [Apache ZooKeeper](https://zookeeper.apache.org/) ensembles. The Stackable Apache ZooKeeper Operator
    is part of the Stackable Data Platform, a curated selection of the best open source data apps like Kafka, Druid, Trino or Spark, all
    working together seamlessly. Based on Kubernetes, it runs everywhere – on prem or in the cloud.

    You can install the operator using [stackablectl or helm](https://docs.stackable.tech/zookeeper/stable/getting_started/installation.html).
    See it in action in one of our [demos](https://stackable.tech/en/demos/) or follow this
    [tutorial](https://docs.stackable.tech/zookeeper/stable/getting_started/first_steps.html).

    N.B. this operator requires the following Stackable internal operators to be installed as well:

    - [Commons Operator](https://github.com/stackabletech/commons-operator)
    - [Secret Operator](https://github.com/stackabletech/secret-operator)
```

File: deploy/olm/23.4.0-rc1/manifests/csv.yaml [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        # version is the spec.versions[].name value defined in the CRD
+        version: v1alpha1
+        # kind is the CamelCased singular value defined in spec.names.kind of the CRD.
+        kind: ZookeeperZnode
+        # human-friendly display name of the CRD for rendering in graphical consoles (optional)
+        displayName: Apache Zookeeper node
+        # a short description of the CRDs purpose for rendering in graphical consoles (optional)
+        description: Represents a Zookeeper node in a cluster
+
+
```
Comment (adwk67) [inline deploy/olm/23.4.0-rc1/manifests/csv.yaml:66]:
```suggestion
  relatedImages:
    - name: zookeeper-operator
      image: docker.stackable.tech/stackable/zookeeper-operator:23.4.0-rc1
```

File: deploy/olm/23.4.0-rc1/manifests/csv.yaml [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+                labels:
+                  app.kubernetes.io/name: zookeeper-operator
+                  app.kubernetes.io/instance: zookeeper-operator
+              spec:
+                serviceAccountName: zookeeper-operator
+                securityContext: {}
+                containers:
+                  - name: zookeeper-operator
+                    securityContext: {}
+                    image: "docker.stackable.tech/stackable/zookeeper-operator:23.4.0-rc1"
```
Comment (adwk67) [inline deploy/olm/23.4.0-rc1/manifests/csv.yaml:217]: do we want digests here already, or leave that for the certification pipeline?

**Review by adwk67 — COMMENTED** [review]
I've left a few comments, which are (only) relevant if we want the bundle definitions to be as close as possible to the files we will use in the PR for certification. We can replace the image tags with digests by setting the relevant pipeline parameter (haven't tried that myself), but I think it's a good idea (for release automation etc.) to keep other things as consistent as possible across the repositories.

File: deploy/olm/23.4.0-rc1/manifests/csv.yaml (hunk 4 above)
Comment (adwk67) [reply deploy/olm/23.4.0-rc1/manifests/csv.yaml:217]: > do we want digests here already, or leave that for the certification pipeline?

On second thoughts, this makes little sense in the bundles as we probably want the flexibility to re-build with the same tag.

Comment (razvan) [reply deploy/olm/23.4.0-rc1/manifests/csv.yaml]: yeah, I agree that we want tags here

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**